libs/community/langchain_community/llms/openllm.py

```diff
@@ -6,247 +6,121 @@
 
     Learn more at: https://github.com/bentoml/openllm
 
     Example running an LLM model locally managed by OpenLLM:
         .. code-block:: python
 
-            from langchain_community.llms import OpenLLM
-            llm = OpenLLM(
-                model_name='flan-t5',
-                model_id='google/flan-t5-large',
-            )
-            llm.invoke("What is the difference between a duck and a goose?")
-
-    For all available supported models, you can run 'openllm models'.
-
-    If you have a OpenLLM server running, you can also use it remotely:
-        .. code-block:: python
-
-            from langchain_community.llms import OpenLLM
-            llm = OpenLLM(server_url='http://localhost:3000')
+            from langchain_community.llms import OpenLLMAPI
+            llm = OpenLLMAPI(server_url='localhost:3000')
             llm.invoke("What is the difference between a duck and a goose?")
     """
 
-    model_name: Optional[str] = None
-    """Model name to use. See 'openllm models' for all available models."""
-    model_id: Optional[str] = None
-    """Model Id to use. If not provided, will use the default model for the model name.
-    See 'openllm models' for all available model variants."""
     server_url: Optional[str] = None
     """Optional server URL that currently runs a LLMServer with 'openllm start'."""
     timeout: int = 30
     """"Time out for the openllm client"""
-    server_type: ServerType = "http"
-    """Optional server type. Either 'http' or 'grpc'."""
-    embedded: bool = True
-    """Initialize this LLM instance in current process by default. Should
-    only set to False when using in conjunction with BentoML Service."""
     llm_kwargs: Dict[str, Any]
     """Keyword arguments to be passed to openllm.LLM"""
 
-    _runner: Optional[openllm.LLMRunner] = PrivateAttr(default=None)
-    _client: Union[
-        openllm.client.HTTPClient, openllm.client.GrpcClient, None
-    ] = PrivateAttr(default=None)
+    _sync_client: openllm.HTTPClient = PrivateAttr(default=None)  #: :meta private:
+    _async_client: openllm.AsyncHTTPClient = PrivateAttr(
+        default=None
+    )  #: :meta private:
 
     class Config:
         extra = "forbid"
 
-    @overload
-    def __init__(
-        self,
-        model_name: Optional[str] = ...,
-        *,
-        model_id: Optional[str] = ...,
-        embedded: Literal[True, False] = ...,
-        **llm_kwargs: Any,
-    ) -> None:
-        ...
-
-    @overload
-    def __init__(
-        self,
-        *,
-        server_url: str = ...,
-        server_type: Literal["grpc", "http"] = ...,
-        **llm_kwargs: Any,
-    ) -> None:
-        ...
-
     def __init__(
         self,
-        model_name: Optional[str] = None,
-        *,
-        model_id: Optional[str] = None,
-        server_url: Optional[str] = None,
+        server_url: str,
         timeout: int = 30,
-        server_type: Literal["grpc", "http"] = "http",
-        embedded: bool = True,
```

Comment:
shoudl we keep around the old args and raise a warning if they're passed in, so that this isn't a breaking change?

Reply:
sure I think we can do that.

Reply:
oh, this is for the new class OpenLLMAPI, but for OpenLLM we can keep backward compat.

Reply:
ah missed the rename :+1